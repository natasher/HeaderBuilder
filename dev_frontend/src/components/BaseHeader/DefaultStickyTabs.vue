<template>
  <ul>

    <li :class = "{ 'active': getStickyFlag === 'default' }"
        @click = "setStickyFlag({ type: 'default' })">
      <a data-type = "default">
        Default
      </a>
    </li>

    <li :class = "{ 'active': getStickyFlag === 'sticky' }"
        @click = "setStickyFlag({ type: 'sticky' })"
        v-if   = "showSticky" >
      <a data-type = "sticky">
        Sticky
      </a>
    </li>

  </ul>
</template>

<script>
import { mapState, mapGetters, mapActions } from 'vuex';

export default {

  name: 'DefaultStickyTabs',

  computed: {
    ...mapState( 'general', {
      getCurrentStoreModule: state => state.currentStoreModule,
    }),
    ...mapState( 'ui', {
      getStickyFlag: state => state.stickyFlag,
    }),
    ...mapGetters( 'ui', [ 'getLayoutPosition' ]),

    showSticky() {
      return this.getCurrentStoreModule === 'DesktopModule'
                && ( this.getLayoutPosition === 'top' || this.getLayoutPosition === 'bottom' )
                || this.getCurrentStoreModule !== 'DesktopModule'
    },
  },

  methods: {
    ...mapActions( 'ui', [ 'setStickyFlag' ]),
  },
};
</script>
