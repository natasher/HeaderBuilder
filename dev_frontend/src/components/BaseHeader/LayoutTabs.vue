<template>
  <div>

    <ul class = "small">

      <li v-for        = "( position, index ) in layoutPositions"
        :key           = "index"
        :data-position = "position"
        @click         = "handleLayoutPositionClick({ position: position })"
        :class         = "{
                            'active': ( getLayoutPosition === position ) ? true : false,
                          }"
        >
          <span class = "icon"></span>
      </li>

    </ul>

  </div>
</template>

<script>
import { mapActions, mapGetters } from 'vuex'

export default {

  name: 'LayoutTabs',

  data() {
    return {
      layoutPositions: [ 'top', 'bottom', 'left', 'right' ],
    }
  },

  computed: {
    ...mapGetters( 'ui', [
      'getLayoutPosition',
    ]),
  },

  methods: {
    ...mapActions( 'ui', [
      'handleLayoutPositionClick',
    ]),

  },

}
</script>
