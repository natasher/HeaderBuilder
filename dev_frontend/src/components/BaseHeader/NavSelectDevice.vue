<template>
  <nav >
    <ul class = "tabs select-device">

      <li @click = "changeToDesktop"
          :class = "{
            'active': ( getCurrentView === 'desktop' ) ? true : false
          }">
        <a data-type = "desktop">
          <span class = "icon">
            <font-icon icon = "monitor" />
          </span>
          <span class = "text">Desktop</span>
        </a>
      </li>

      <li @click = "changeToTablet"
          :class = "{
            'active': ( getCurrentView === 'tablet' ) ? true : false
          }">
        <a data-type = "tablet">
          <span class = "icon">
            <font-icon icon = "mobile-line" />
          </span>
          <span class = "text">Tablet</span>
        </a>
      </li>

      <li @click = "changeToMobile"
          :class = "{
            'active': ( getCurrentView === 'mobile' ) ? true : false
          }">
        <a data-type = "mobile">
          <span class = "icon">
            <font-icon icon = "mobile" />
          </span>
          <span class = "text">Mobile</span>
        </a>
      </li>

    </ul>
  </nav>
</template>

<script>
import { mapActions, mapGetters, mapState } from 'vuex'
import FontIcon from '../BaseGeneric/FontIcon.vue'
import { infoMobileInactiveItems } from '../../store/dialogs'

export default {

  name: 'NavSelectDevice',

  components: {
    FontIcon,
  },

  computed: {
    ...mapState( 'general', {
      getCurrentView: state => state.currentView,
    }),
    ...mapGetters( 'ui', [ 'getGridStatus' ]),
  },

  methods: {
    ...mapActions( 'grid', [
      'clone',
    ]),

    ...mapActions( 'inactiveItems', [
      'clear',
    ]),

    ...mapActions( 'ui', [
      'changeDevice',
    ]),

    changeToDesktop: function() {
      this.changeDevice({ device: 'desktop' })
    },

    changeToTablet: function() {
      this.changeDevice({ device: 'tablet' })
        .then(() => {
          this.$nextTick(() => {

            if (this.getGridStatus === 'auto') {
              this.clone({ from: 'DesktopModule' })
            }

          })
        })
    },

    changeToMobile: function() {
      this.changeDevice({ device: 'mobile' })
        .then(() => {
          this.$nextTick(() => {

            if (this.getGridStatus === 'auto') {
              this.clear()
              this.clone({ from: 'DesktopModule' })

              this.$modal.show('MfnGenericModal', {
                title       : 'Info',
                text        : infoMobileInactiveItems,
                closeOnClick: true,
              })
            }

          })
        })
    },

  },
}
</script>
