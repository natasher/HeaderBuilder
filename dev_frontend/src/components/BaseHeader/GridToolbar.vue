<template>
 <wrapper-clearfix class="general">

      <default-sticky-tabs class="tabs" />

      <wrapper-section title="Show rows"
        class="rows">
          <base-switch
            label         = "Action bar"
            :isActive     = "getActionBarActive"
            @click.native = "toggleActionBar" />

          <base-switch
            label         = "Second row"
            :isActive     = "getSecondRowActive"
            @click.native = "toggleSecondRow" />
      </wrapper-section>

      <wrapper-section title = "Layouts"
        class = "layouts"
        v-if = "getCurrentStoreModule === 'DesktopModule'">
        <layout-tabs class = "select-position" />
      </wrapper-section>

  </wrapper-clearfix>
</template>

<script>
import { mapState, mapGetters, mapActions } from 'vuex'

import WrapperClearfix   from '../BaseGeneric/WrapperClearfix.vue'
import WrapperSection    from '../BaseGeneric/WrapperSection.vue'
import BaseSwitch        from '../BaseGeneric/BaseSwitch.vue'
import DefaultStickyTabs from './DefaultStickyTabs.vue'
import LayoutTabs        from './LayoutTabs.vue'

export default {

  name: 'GridToolbar',

  components: {
    WrapperClearfix,
    WrapperSection,
    BaseSwitch,
    DefaultStickyTabs,
    LayoutTabs,
  },

  computed: {
    ...mapState( 'general', {
      getCurrentStoreModule: state => state.currentStoreModule,
    }),
    ...mapGetters( 'ui', [
      'getActionBarActive',
      'getSecondRowActive'
    ]),
  },

  methods: {
    ...mapActions( 'ui', [
      'toggleActionBar',
      'toggleSecondRow'
    ]),

  },
}
</script>
