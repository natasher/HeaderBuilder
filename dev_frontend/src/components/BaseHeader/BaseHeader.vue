<template>
  <header class = "header">

    <div class = "left">
      <mfn-logo />
      <base-title title = "Header Builder" />
    </div>

    <div class = "center">

      <nav-select-device />

    </div>

    <div class="right">

      <a id            = "header-builder-submit"
        class          = "mfn-button mfn-button-primary"
        @click.prevent = "postData">
        Save changes
      </a>

    </div>

  </header>

</template>

<script>
import { mapActions }  from 'vuex'
import MfnLogo         from '../BaseGeneric/MfnLogo.vue'
import BaseTitle       from '../BaseGeneric/BaseTitle.vue'
import FontIcon        from '../BaseGeneric/FontIcon.vue'
import NavSelectDevice from './NavSelectDevice.vue'

export default {

  name: 'BaseHeader',

  components: {
    MfnLogo,
    BaseTitle,
    FontIcon,
    NavSelectDevice,
  },

  methods: {
    ...mapActions( 'endpoint', [
      'postDataToWordpress',
    ]),

    postData: function () {
      this.postDataToWordpress({ $modal: this.$modal })
    },
  },

}
</script>
