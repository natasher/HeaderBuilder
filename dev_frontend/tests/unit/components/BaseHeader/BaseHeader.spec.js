import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex          from 'vuex'
import flushPromises from 'flush-promises'
import merge         from 'lodash.merge'
import BaseHeader    from '../../../../src/components/BaseHeader/BaseHeader.vue'

const localVue = createLocalVue()
localVue.use( Vuex )

function createStore(overrides) {
  const defaultStoreConfig = {
    modules: {
      endpoint: {
        namespaced: true,
        actions: {
          postDataToWordpress: jest.fn(() => Promise.resolve()),
        }
      }
    }
  }

  return new Vuex.Store(
    merge( defaultStoreConfig, overrides )
  )
}

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue,
    store: createStore(),
    mocks: {
      $modal: jest.fn()
    },
  }

  return shallowMount( BaseHeader, merge( defaultMountingOptions, overrides ) )
}

describe('BaseHeader.vue', () => {

  test('should match snapshot', () => {
    const wrapper = createWrapper()

    expect( wrapper.element ).toMatchSnapshot()
  })

  test('should dispatch `postDataToWordpress` action when `Save` button is clicked', async () => {
    const store   = createStore()
    const wrapper = createWrapper({ store, })

    const action = jest.spyOn( store._actions[ 'endpoint/postDataToWordpress' ], [0])
    wrapper.find('#header-builder-submit').trigger('click')
    await flushPromises()

    expect( action ).toBeCalledWith({ $modal: wrapper.vm.$modal })
  })

})