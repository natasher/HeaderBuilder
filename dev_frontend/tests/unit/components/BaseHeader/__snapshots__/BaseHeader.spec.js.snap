// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseHeader.vue should match snapshot 1`] = `
<header
  class="header"
>
  <div
    class="left"
  >
    <mfnlogo-stub />
     
    <basetitle-stub
      title="Header Builder"
    />
  </div>
   
  <div
    class="center"
  >
    <navselectdevice-stub />
  </div>
   
  <div
    class="right"
  >
    <a
      class="mfn-button mfn-button-primary"
      id="header-builder-submit"
    >
      
      Save changes
    
    </a>
  </div>
</header>
`;
